connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java

```diff
@@ -159,6 +160,18 @@ public void translate(InventoryTransactionPacket packet, GeyserSession session)
                         Position pos = new Position(packet.getBlockPosition().getX(), packet.getBlockPosition().getY(), packet.getBlockPosition().getZ());
                         ClientPlayerActionPacket breakPacket = new ClientPlayerActionPacket(action, pos, BlockFace.values()[packet.getFace()]);
                         session.sendDownstreamPacket(breakPacket);
+
+                        // Add a check if mobile players try to break blocks too far away
+                        // 6f is an approximation
+                        if (packet.getPlayerPosition().distance(packet.getBlockPosition().toFloat()) > 6f) {
+                            if (session.getConnector().getConfig().isCacheChunks()) {
+                                ChunkUtils.updateBlock(session, session.getConnector().getWorldManager().getBlockAt(session,
+                                        packet.getBlockPosition().getX(), packet.getBlockPosition().getY(), packet.getBlockPosition().getZ()), packet.getBlockPosition());
+                            } else {
+                                session.sendMessage("The block you just destroyed may not have been sent to the server.");
```

Comment:
More description should be added to this (such as the long reach not being supported) so players who receive this message know why.

Reply:
Also casual player are unlikely to know about the client/server architecture (they probably know what a server is but it still could be confusing)

Reply:
Good point, Circuit.